src/System.Private.ServiceModel/tools/IISHostedWcfService/App_code/IWcfService.cs

```diff
@@ -33,6 +33,16 @@ public interface IWcfService
         void TestFaultInt(int faultCode);
 
         [OperationContract]
+        [FaultContract(typeof(FaultDetail), Action = "http://tempuri.org/IWcfService/TestFaultFaultDetailFault", Name = "FaultDetail", Namespace = "http://www.contoso.com/wcfnamespace")]
+        [FaultContract(typeof(FaultDetail2), Action = "http://tempuri.org/IWcfService/TestFaultFaultDetailFault2", Name = "FaultDetail2", Namespace = "http://www.contoso.com/wcfnamespace")]
+        void TestFaults(string faultMsg, bool flag);
```

Comment:
Nit: it would be better to have a more descriptive parameter name, like 'bool throwFaultDetail'